### PR TITLE
test: Adds storybook to SearchInput component

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
- Adds storybook to `SearchInput` component
- Moves `SearchInput` tests to component's folder

### TEST PLAN
1 - Execute `SearchInput` tests
2 - All tests should pass

@rusackas 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/SearchInput/SearchInput.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        onChange={e => setCurrentValue(e.target.value)}
+        onClear={() => setCurrentValue('')}
+      />
+    </div>
+  );
+};
+
+InteractiveSearchInput.args = {
+  value: 'Test',
+  placeholder: 'Enter a text',
```
Comment (rusackas) [inline superset-frontend/src/components/SearchInput/SearchInput.stories.tsx:46]:
```suggestion
  placeholder: 'Enter some text',
```

File: superset-frontend/src/components/SearchInput/SearchInput.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  component: SearchInput,
+};
+
+export const InteractiveSearchInput = ({
+  value,
+  ...rest
+}: SearchInputProps) => {
+  const [currentValue, setCurrentValue] = useState(value);
+  return (
+    <div style={{ width: 230 }}>
```
Comment (rusackas) [inline superset-frontend/src/components/SearchInput/SearchInput.stories.tsx:33]: Just a curiosity - 230 seems... specific. Is there a reason for that, e.g. a layout feature (full-width, etc) that should be tweakable or demonstrated by the story?

**Review by rusackas — APPROVED** [review]
Just a nit/qustion... approving, and will circle back for the merge one way or another.

File: superset-frontend/src/components/SearchInput/SearchInput.stories.tsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/SearchInput/SearchInput.stories.tsx:33]: Not really, it's just to look more beautiful when viewing the story. The story host takes full width and I don't want the `Select` to stretch 😉.

**Pull request merged**